### PR TITLE
MINGASPRICE Opcode draft

### PR DESCRIPTION
**Pull request by manfr3d**
Peer review of draft and GAS COST of Opcode (now 2).

### CODE REVIEW

File: IPs/RSKIP-64 [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+* Network congestion of the network. Transaction congestion can be detected by measuring a higher than usual Minimum Gas Price.
+* Financial behavior attacks, based on statistics, by making the economically infeasible, i.e. the attack cannot make money by statistically manipulating the contract.
+
+For the first type of scenario the contract may be designed to reduce its activity if the minimum gas price is too high. For example, if we are collecting 0.01 RSK donations but the transaction will cost 0.02 RSK then the contract will revert all transactions (REVERT Opcode, revert() or require() on Solidity) refunding most of the gas [4], until a most economically favorable network is detected.
```
Comment (SergioDemianLerner) [inline IPs/RSKIP-64]: The native coin name is SBTC, not RSK.

File: IPs/RSKIP-64 [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+For the second type of scenario the opcode will help with the self-control of smart contracts that change their dynamic behavior based on number of events per measure of time. In the last case, knowing the minimum gas price can help the smart contract estimate the cost of an attack to manipulate its behavior, thus limiting the attack, making it economically infeasible. This use case is a generalization of the case in which a smart token controling its supply be measuring the number of transactions per day is attack by a user trying to manipulate the number of transaction to inflate the token supply.
+
+# Specification
+
+A new opcode is added: MINGASPRICE
```
Comment (SergioDemianLerner) [inline IPs/RSKIP-64]: Specify the opcode numerical value

File: IPs/RSKIP-64 (hunk 1 above)
Comment (jorlicki) [reply IPs/RSKIP-64]: Done.

**Comment by jorlicki** [comment]
Done improvements.

File: IPs/RSKIP64.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+No arguments are pushed by caller. The minimum gas price value is pushed on the stack after execution of this opcode.
+
+The gas cost of opcode MINGASPRICE is 2.
+The opcode number is 0xAC = 172.
+
+The gas is consumed immediately when the opcode is executed.
+
+
+## Rationale
+
```
[the remark this answers is not in the document]
Comment (SergioDemianLerner) [reply IPs/RSKIP64.md]: I'm thinking on a meta-RSK where we define a scheme for adding more opcodes to RSK without interfering with the opcodes in Ethereum. 
We have several choices :  
1. use the memory mapped configuration register https://github.com/rsksmart/RSKIPs/blob/master/IPs/RSKIP51.md
to change the VM version and then re-interpret opcode values
2. Use CALL to special addresses as gateways. THIS SEEM TO BE CLEANER

**Comment by SergioDemianLerner** [comment]
This opcode was superseded by the RSKIP that proposes a precompile that provides much more information about the last header and the previous 4k headers. (RSKIP number?).
To keep this RSKIP for future reference, we should reassign a number to it, because the number 64 was reused.

**Pull request merged**